### PR TITLE
 add http server for nodeservice

### PR DESCRIPTION
**Pull request by mayurkale22**

### CODE REVIEW

File: interoptest/src/nodejsservice/src/testservice/service-hopper.js [hunk 1]
```diff
@@ -94,6 +116,57 @@ function grpcServiceHop (host, port, request) {
   });
 }
 
+function toBuffer (testRequest) {
```
Comment (rghetia) [inline interoptest/src/nodejsservice/src/testservice/service-hopper.js]: common code. can this be put in a utility file?

File: interoptest/src/nodejsservice/src/testservice/service-hopper.js [hunk 2]
```diff
@@ -94,6 +116,57 @@ function grpcServiceHop (host, port, request) {
...
+      let req = http.request(options, res => {
+        let data = [];
+        res.on('data', chunk => { data.push(chunk); });
+        res.on('end', () => {
+          if (res.statusCode === 200) {
+            const bytes = new Uint8Array(Buffer.concat(data));
+            response = interop.TestResponse.deserializeBinary(bytes);
+            resolve(response);
+          } else {
+            resolve(setFailureStatus(response, 'Http Service Hopper Error'));
```
Comment (rghetia) [inline interoptest/src/nodejsservice/src/testservice/service-hopper.js]: include statusCode and perhaps error string if available.

File: interoptest/src/nodejsservice/src/testservice/http-server.js [hunk 3]
```diff
@@ -40,9 +41,34 @@ function start (httpPort, httpHost) {
...
+        const testRequest = interop.TestRequest.deserializeBinary(bytes);
+        (async () => {
+          const testResponse = await serviceHopper.serviceHop(testRequest);
+          console.log(`http hopper:${JSON.stringify(testResponse.toObject())}`);
+          response.writeHead(200, PROTOBUF_HEADER);
+          response.write(toBuffer(testResponse));
+          response.end();
+        })();
+      });
+    }
```
Comment (rghetia) [inline interoptest/src/nodejsservice/src/testservice/http-server.js]: if url is not URL_ENDPOINT then what?

File: interoptest/src/nodejsservice/src/testservice/service-hopper.js (hunk 1 above)
Comment (mayurkale22) [reply interoptest/src/nodejsservice/src/testservice/service-hopper.js]: Done

File: interoptest/src/nodejsservice/src/testservice/service-hopper.js (hunk 2 above)
Comment (mayurkale22) [reply interoptest/src/nodejsservice/src/testservice/service-hopper.js]: SGTM, done. 

Examples:
```Http Service Hopper Error: connect ECONNREFUSED 127.0.0.1:10309```
```Http Service Hopper Error 400:Bad Request```
```GRPC Service Hopper Error : 14 UNAVAILABLE: Connect Failed```
```Http Socket Error : Port should be >= 0 and < 65536. Received 103009```

File: interoptest/src/nodejsservice/src/testservice/http-server.js (hunk 3 above)
Comment (mayurkale22) [reply interoptest/src/nodejsservice/src/testservice/http-server.js]: Added ```400(BadRequest)```, if url is not URL_ENDPOINT.

**Pull request merged**